src/Opaleye/Internal/HaskellDB/Sql/Print.hs

```diff
@@ -97,11 +97,14 @@ ppDelete (SqlDelete table criteria) =
 
 
 ppInsert :: SqlInsert -> Doc
-ppInsert (SqlInsert table names values)
+ppInsert (SqlInsert table names values onConflict)
     = text "INSERT INTO" <+> ppTable table
       <+> parens (commaV ppColumn names)
       $$ text "VALUES" <+> commaV (parens . commaV ppSqlExpr)
                                   (NEL.toList values)
+      <+> text (case onConflict of
+        Just _ -> "ON CONFLICT DO NOTHING"
```

Comment:
Please avoid a wildcard pattern here and use `DoNothing`.  The downside of using a wildcard pattern is that we will not get a compiler warning if we forget to add a future clause for `OnConflict ...`.